internal/core/services/workers_manager/workers_manager_test.go

```diff
@@ -0,0 +1,228 @@
+package workers_manager
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
+
+	opflow "github.com/topfreegames/maestro/internal/adapters/operation_flow/mock"
+	opstorage "github.com/topfreegames/maestro/internal/adapters/operation_storage/mock"
+	schedulerStorageMock "github.com/topfreegames/maestro/internal/adapters/scheduler_storage/mock"
+	configMock "github.com/topfreegames/maestro/internal/config/mock"
+	"github.com/topfreegames/maestro/internal/core/entities"
+	"github.com/topfreegames/maestro/internal/core/ports/errors"
+	"github.com/topfreegames/maestro/internal/core/services/operation_manager"
+	"github.com/topfreegames/maestro/internal/core/workers"
+)
+
+func TestStart(t *testing.T) {
+
+	t.Run("with success", func(t *testing.T) {
+
+		core, recorded := observer.New(zap.InfoLevel)
+		zl := zap.New(core)
+		zap.ReplaceGlobals(zl)
+
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		configs := configMock.NewMockConfig(mockCtrl)
+		schedulerStorage := schedulerStorageMock.NewMockSchedulerStorage(mockCtrl)
+		operationFlow := opflow.NewMockOperationFlow(mockCtrl)
+		operationStorage := opstorage.NewMockOperationStorage(mockCtrl)
+		operationManager := operation_manager.New(operationFlow, operationStorage)
+
+		configs.EXPECT().GetInt(syncOperationWorkersIntervalPath).Return(10)
+		configs.EXPECT().GetInt(workers.OperationWorkerIntervalPath).Return(10)
+		schedulerStorage.EXPECT().GetAllSchedulers(context.Background()).Return([]*entities.Scheduler{
+			{
+				Name:            "zooba-us",
+				Game:            "zooba",
+				State:           entities.StateCreating,
+				RollbackVersion: "1.0.0",
+				PortRange: &entities.PortRange{
+					Start: 1,
+					End:   10000,
+				},
+			},
+		}, nil)
+
+		workersManager := NewWorkersManager(configs, schedulerStorage, *operationManager)
+
+		err := workersManager.Start(context.Background())
+		require.NoError(t, err)
+
+		require.Contains(t, workersManager.CurrentWorkers, "zooba-us")
+
+		assertLogMessages(t, recorded, map[zapcore.Level][]string{
```

Comment:
I'm not sure about making assertions on the logs, it seems a bit fragile.

Reply:
Once we have no metrics already defined, I'm keeping the logs assertions.

As soon as we have metrics in place, we can remove it.